File: Tasks/src/decorator/PrinterTaskRunner.java
package decorator;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class PrinterTaskRunner extends AbstractTaskRunner {

   //Metoda statica pentru a formata ora
    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");

    //Constructor
    public PrinterTaskRunner(TaskRunner taskRunner) {
        //Apeleaza constructorul clasei de baza AbstractTaskRunner
        super(taskRunner);
    }

    //Suprascrie metoda executeOneTask()
    @Override
    public void executeOneTask() {
        //Apeleaza metoda executeOneTask() din clasa de baza AbstractTaskRunner
        super.executeOneTask();
        //System.out.println("Task executed at: " + LocalDateTime.now().format(TIME_FORMATTER));
        System.out.println("Task executed at: " + LocalDateTime.now().format(TIME_FORMATTER));
    }
}

File: Tasks/runners/decorator/PrinterTaskRunner.cs
namespace Lab10.runners;

public class PrinterTaskRunner : AbstractTaskRunner
{
    //Clasa PrinterTaskRunner extinde clasa AbstractTaskRunner (decorator)
    
    //Formatter static pentru data
    private static readonly string DATE_FORMAT = "yyyy-MM-dd HH:mm";
    
    //Constructor
    public PrinterTaskRunner(TaskRunner taskRunner) : base(taskRunner)
    {
        if (taskRunner == null)
        {
            throw new ArgumentNullException(nameof(taskRunner), "TaskRunner cannot be null");
        }
    }

    public override void executeOneTask()
    {
        base.executeOneTask();
        Console.WriteLine("Task executed at: " + DateTime.Now.ToString(DATE_FORMAT));
    }
}